src/libs/ajax.js

```
@@ -1110,9 +1110,23 @@ const DataRepo = signal => ({
       details: async () => {
         const res = await fetchDataRepo(`repository/v1/snapshots/${snapshotId}`, _.merge(authOpts(), { signal }))
         return res.json()
+      },
+      exportSnapshot: async () => {
+        const res = await fetchDataRepo(`repository/v1/snapshots/${snapshotId}/export`, _.merge(authOpts(), { signal }))
```

Worth noting: there are other URL parameters that can be passed in to export:
- `exportGsPaths`: can be ignored for now and defaults to false
- `validatePrimaryKeyUniqueness`: this is set to true for now but if we're working with azure snapshots, that needs to be set to false (for now) or the export job will fail.  The cloud provider for the snapshot is returned as part of the snapshot now (as of yesterday)
Does it make sense to make that change now or as a followon?

---

I think followon, if at all. I think there's conversations we need to have internally to the catalog on whether we want to support what here, or just use sensible defaults and obfuscate it from the non-power users.

---

Right.  I do think we'll want the ability to set it in the call (and not expose it to user...but just to make sure it works from Catalog). By setting that, we'll be able to support azure snapshots starting today (which is super cool)

---

Lets do it as a follow on, because we haven't yet trial ran azure snapshots in the catalog, so we may need to make some metadata changes to be able to detect it. @tcjiang9 this may relate to schema definitions.

---

I think we should add `validatePrimaryKeyUniqueness=false` for now, to support azure, and open a ticket in the backlog to disable this in the future once azure snapshots support this operation. In the case of a snapshot that's been created for sharing, I'd hope that the data owner has already verified that the data is correct and there's no primary key conflict in the data.

Although if primary key validation doesn't work at all for azure, I'm not sure why an option is needed. Is there a case where a caller would prefer the operation to fail instead of skipping validation for azure?

---

ok sounds good!